cmd/create/cluster/cmd.go

```diff
@@ -1194,22 +1198,21 @@ func run(cmd *cobra.Command, _ []string) {
 				})
 			}
 		}
-		// Validate the role names are available on AWS
-		for _, role := range operatorIAMRoleList {
-			name, err := aws.GetResourceIdFromARN(role.RoleARN)
-			if err != nil {
-				r.Reporter.Errorf("Error validating role arn. Arn: %v. Error: %v", role.RoleARN, err)
-				os.Exit(1)
-			}
-
-			err = awsClient.ValidateRoleNameAvailable(name)
-			if err != nil {
-				r.Reporter.Errorf("Error validating role: %v", err)
+		isByoOidcSet, oidcEndpointUrl, oidcPrivateKeySecretArn = handleByoOidcOptions(r, cmd, isSTS)
+		err = validateOperatorRolesAvailabilityUnderUserAwsAccount(awsClient, operatorIAMRoleList)
```

Comment:
If it's a `ByoOidc` flow, do we still have to check that the operator role's name is available?

Reply:
Not really, it can not exist or be in use and it should pick it up either way without any issues. At some point in the future we'll allow to actually create them using the prefix prior to creating the cluster. By then we might want to enforce they pre exist.